### DISCUSSION TITLE
unable to generate METviewer plot

### DISCUSSION THREAD

**Discussion by jwraby** [dtcenter/METplus#2000] [Plot Generation] [answered]
While conducting initial testing of the recently installed version V4.1.0 of METviewer, discovered that I was unable to generate a plot from the xml file which was generated by our previously installed version of METviewer (V2.11). Used the "Load XML" button to load the xml file via the web interface. Clicking on "Generate Plot" reports "No image available" in the "Plot" tab. Clicked on the "Log" tab to see the error dialog, but there is no dialog. A new plot file name is displayed. Attempted to download example plot xml files from the V4.1.0 User's Guide to see if I can generate the plots, but it fails reporting a 404 error. The URL for the User's Guide is as follows:
https://metviewer.readthedocs.io/en/latest/Users_Guide/barplots.html

Should I expect that the V4.1.0 version would be able to plot an xml file generated by an older version of METviewer? 
Do you have a download site where I can obtain example plot xml files for use in testing? 
Do you have any thoughts on why I see nothing in the "Log" tab?

Thanks.

**Comment by georgemccabe** [comment]
Hi @jwraby, many of our developers are still out on vacation this week. I will need to coordinate with them when they return to answer some of your questions. In the meantime, here is the [example XML file](https://raw.githubusercontent.com/dtcenter/METviewer/main_v5.0/docs/_static/xml/bar_plots_xml.xml) that is referenced in the broken link. I would not be surprised if a file generated by an older version could not be read in the newer version. Hopefully this example will help you make the necessary changes.

**Reply by jwraby** [reply]
Thanks for the example XML file!

**Reply by jwraby** [reply]
George - I tried to download the file but I think it is being blocked by the Army firewall. Could you send it as an email attachment? john.w.raby2.civ@army.mil

**Reply by georgemccabe** [reply]
@jwraby, I just emailed the file to you.

**Reply by jwraby** [reply]
Email received and file successfully downloaded. Thanks.

**Comment by venitahagerty** [comment, marked as answer]
Hi @jwraby, I work on met_db_load.py, so I know a little about METviewer. I know that we can still read older version of MET files, so I expect that METviewer generally can still read older XML files. Do the other tabs such as XML and SQL show information? Do you know that you have the data in the specified database that matches the query used by the plot? Are you able to generate a plot with a new XML file, and do those tabs show information?

**Reply by jwraby** [reply]
I looked at all the other tabs and there is no information in any of them except for the "Plot" tab which shows "No image available". Good question about the availability of the database. The answer is negative. The new METviewer installation is on a new machine and the data base from which the original plot was created does not exist. I will download the example XML file provided by George and see if I can generate that plot and then provide and update on the results in answer to your last question.

**Reply by jwraby** [reply]
I tried to generate a plot using and example file provided by George. Used the same procedure (Load XML, then Generate Plot). A lot of the settings on the left side of the web interface changed as well as the name of the database, but I get the same results (no plot and no information in all of the other tabs).

**Reply by venitahagerty** [reply]
If all the tabs are blank, it's possible METviewer was not installed correctly. Also, METviewer generates plots from data in the database named in the XML file. If you have not loaded data that matches your plot parameters into the named database, you can't generate a plot from METviewer,

**Reply by jwraby** [reply]
Thanks for your analysis of these results and your recommendations. Does the User's Guide provide the instructions for database creation and loading? I have done this for V2.11 using load spec files.

**Reply by venitahagerty** [reply]
The database loading procedure is still very similar, using a load spec XML file. The Java loader has been replaced with the METdatadb (now METdataio) Python loader, METdbLoad. Here is a link to the Version 2 (numbered differently than METviewer) User's Guide:

[METdatadb User's Guide, Version 2](https://metdatadb.readthedocs.io/en/v2.0.0-beta1/Users_Guide/index.html
)

A database must be created within the database server with a name like mv_ens (starting with the prefix mv), and tables set up with the schema. Section 2 describes that process, and sections 3 and 4 describe the tags for the load spec XML file.

**Reply by jwraby** [reply]
Thanks for this information. Regarding the replacement of the Java loader, does this apply to METviewer V4.1.0? Thanks.

**Reply by venitahagerty** [reply]
METdatadb became available with METviewer 4.0.0. I think the Java loader was still available in METviewer 4.1.0. For METviewer 5.0.0, changes for loading new MET data were only made in METdatadb/METdataio.

**Reply by jwraby** [reply]
Thanks for answering my question. My system admin reports that he has successfully created some test databases. I plan to work on loading them with data as the next step.
